src/whylogs/app/logger.py

```diff
@@ -189,19 +249,21 @@ def close(self) -> Optional[DatasetProfile]:
         if self.with_rotation_time is None:
             self.flush()
         else:
-            self.rotate_time()       
+            self._rotate_time()     
             _ = self._profiles.pop()
 
         self._active = False
-        profile = self._profiles[-1]
+        profile = self._profiles[-1]["full_profile"]
         self._profiles = None
         return profile
 
     def log(
         self,
-        features: typing.Dict[str, any] = None,
+        features: Dict[str, any] = None,
```

Comment:
shouldn't this be `Optional[...]` (not related to the current PR but still wondering)

Reply:
yes

Reply:
```suggestion
        features: Optional[Dict[str, any]] = None,
```